hs_core/search_indexes.py

```diff
@@ -44,65 +48,130 @@ def get_model(self):
         return BaseResource
 
     def index_queryset(self, using=None):
-        return self.get_model().objects.all()
+        return self.get_model().objects.filter(Q(raccess__discoverable=True) | Q(raccess__public=True))
 
     def prepare_title(self, obj):
-        return obj.metadata.title.value
+        if hasattr(obj.metadata.title, 'value'):
+            return obj.metadata.title.value
```

Comment:
@hyi I used `if hasattr(obj, 'metadata')`, but it reported some error when I created a new resource. But just took it back and tested, it seems it works correctly now.